### PR TITLE
Change logout to use post method and add DeleteCookies option

### PR DESCRIPTION
**Pull request by dorsha**
## Description
- Change logout to logoutAll and use POST method
- Add DeleteCookies method

## Must
- [x] Tests

### CODE REVIEW

**Review by asafshen — APPROVED** [review]

File: descope/auth/sdk.go [hunk 1]
```diff
@@ -252,15 +252,24 @@ type Authentication interface {
 	// the specified roles for a specific tenant.
 	ValidateTenantRoles(token *Token, tenant string, roles []string) bool
 
-	// Logout - Use to perform logout from all active devices. This will revoke the given tokens
+	// Logout - Use to perform logout the session user. This will revoke the given token
 	// and if given options will also remove existing session on the given response sent to the client.
 	// Use the ResponseWriter (optional) to apply the cookies to the response automatically.
 	// This is a shortcut for LogoutWithOptions(r, WithResponseOption(w))
 	Logout(request *http.Request, w http.ResponseWriter) error
```
[the remark this answers is not in the document]
Comment (dorsha) [reply descope/auth/sdk.go]: Changed description.

File: descope/auth/auth.go [hunk 2]
```diff
@@ -75,11 +75,40 @@ func (auth *authenticationService) WebAuthn() WebAuthn {
...
+	if request == nil {
+		return errors.MissingRequestError
+	}
+
+	cookies := make([]*http.Cookie, 0)
+	sessionCookie, _ := request.Cookie(SessionCookieName)
+	refreshCookie, _ := request.Cookie(RefreshCookieName)
+
+	if sessionCookie != nil {
+		sessionCookie.MaxAge = -1
```
Comment (asafshen) [inline descope/auth/auth.go:92]: I don't think max age `-1` is a valid syntax, from what I know `maxAge=0` is the protocol 

https://stackoverflow.com/questions/15932957/difference-between-0-and-negative-value-for-setmaxage-for-cookie

Comment (dorsha) [reply descope/auth/auth.go:92]: In go it is -1.

Comment (dorsha) [reply descope/auth/auth.go:92]: image

File: examples/ginwebapp/main.go [hunk 3]
```diff
@@ -126,7 +126,7 @@ func main() {
 	authorized := r.Group("/")
 	authorized.Use(descopegin.AuthneticationMiddleware(client.Auth, nil, nil))
 	authorized.GET("/private", handleIsHealthy)
-	authorized.GET("/logout", handleLogout)
+	authorized.GET("/logout", handleLogout) // Logout from all user's active sessions
```
Comment (asafshen) [inline examples/ginwebapp/main.go:129]: maybe we want to change the logout in our examples to  `POST` as well?

File: descope/auth/auth.go (hunk 2 above)
Comment (asafshen) [reply descope/auth/auth.go:92]: nice 👍

**Review by asafshen — DISMISSED** [review]
👍 

one comment about examples `GET` vs `POST`

File: examples/ginwebapp/main.go (hunk 3 above)
Comment (dorsha) [reply examples/ginwebapp/main.go]: done

**Review by asafshen — APPROVED** [review]
💯

**Pull request merged**